File: D/1704-determine-if-string-halves-are-alike/main.go
package main

import "fmt"

func halvesAreAlike(s string) bool {
	left, right := 0, 0
	for i := 0; i < len(s)/2; i++ {
		if byte(s[i]) == 'a' || byte(s[i]) == 'e' || byte(s[i]) == 'i' || byte(s[i]) == 'o' || byte(s[i]) == 'u' || byte(s[i]) == 'A' || byte(s[i]) == 'E' || byte(s[i]) == 'I' || byte(s[i]) == 'O' || byte(s[i]) == 'U' {
			left++
		}
		if byte(s[len(s)-1-i]) == 'a' || byte(s[len(s)-1-i]) == 'e' || byte(s[len(s)-1-i]) == 'i' || byte(s[len(s)-1-i]) == 'o' || byte(s[len(s)-1-i]) == 'u' || byte(s[len(s)-1-i]) == 'A' || byte(s[len(s)-1-i]) == 'E' || byte(s[len(s)-1-i]) == 'I' || byte(s[len(s)-1-i]) == 'O' || byte(s[len(s)-1-i]) == 'U' {
			right++
		}
	}
	return left == right
}

func main() {
	fmt.Println()
}

File: D/1704-determine-if-string-halves-are-alike/main.rs
struct Solution {}

use std::collections::HashMap;
impl Solution {
  pub fn halves_are_alike(s: String) -> bool {
    let bb = s.as_bytes().to_vec();
    let keys: Vec<u8> = vec![
      'a' as u8, 'e' as u8, 'i' as u8, 'o' as u8, 'u' as u8, 'A' as u8, 'E' as u8, 'I' as u8,
      'O' as u8, 'U' as u8,
    ];
    let mut idx_m: HashMap<u8, usize> = HashMap::new();
    for i in 0..keys.len() {
      idx_m.insert(keys[i], i);
    }

    let size = bb.len() / 2;
    let mut left: i32 = 0;
    let mut right: i32 = 0;
    for i in 0..size {
      let offset = idx_m.get(&bb[i]);
      if offset.is_some() {
        left += 1;
      }

      let offset2 = idx_m.get(&bb[i + size]);
      if offset2.is_some() {
        right += 1;
      }
    }

    // println!("{:?} {:?}", left, right);
    let ret = left == right;
    ret
  }
}

fn main() {
  println!("{}", Solution::halves_are_alike("abcd".to_string()))
}
